### ISSUE TITLE
bug(windows): Plugging in my Ethernet adapter while on Wi-Fi still breaks it

### ISSUE THREAD

**Issue by ReactorScram** [CloudFire-LLC/cloudfire#3355]
**Describe the bug**
Even after #3335 merged, plugging in Ethernet messes something up.

**To Reproduce**
1. Connect Wi-Fi only
2. Start Firezone
3. `curl https://ifconfig.net/ip`, it works
4. Connect Ethernet
5. Curl again, doesn't work anymore

**Expected behavior**
Expected adding a 2nd connection to not disrupt Firezone

**Screenshots / Logs**
Log on internal Slack: https://firezonehq.slack.com/archives/C067DSY7TFX/p1706030258161169

On lines 127 (`"time":"2024-01-23T17:16:11.4607156Z"`) and 131 `haveInternet` changes to false and then back to true when I connect the Ethernet. So we can detect that something happens, but we aren't automatically recovering.

**Platform (please complete the following information):**
 - Firezone Version 42a05e86e
 - OS: Windows 11

**Additional context**

**Comment by jamilbk** [comment]
refs #3343

**Comment by jamilbk** [comment]
Closing in favor of #3343

**Comment by ReactorScram** [comment]
Not seeing this replicate on 127b97e after the big snownet merge 🥳
